src/systemctl/systemctl.c

```diff
@@ -3556,10 +3557,14 @@ static int start_special(int argc, char *argv[], void *userdata) {
         if (r < 0)
                 return r;
 
-        if (a == ACTION_REBOOT && argc > 1) {
-                r = update_reboot_parameter_and_warn(argv[1], false);
-                if (r < 0)
-                        return r;
+        if (a == ACTION_REBOOT) {
+                if (argc > 1 || arg_reboot_arg != NULL) {
+                        if (!arg_reboot_arg)
+                                log_notice("--reboot-arg= is recommended; the positional argument is being deprecated.");
```

Comment:
maybe log if both are used together, since that makes no sense:

```c
const char *a;

if (argc > 1) {
        if (arg_reboot_arg)
                return log_error_errno(SYNTHETIC(EINVAL), "Both --reboot-argument= and positional argument passed to reboot command, refusing.");

        log_notice("Positional argument to reboot command is deprecated, please use --reboot-argument= instead. Accepting anyway.");
        a = argv[1];
} else
        a = arg_reboot_argument;

if (a) {
        r = update_rebo…
```

or something like that?

Reply:
You're right. I thought of giving priority to either (allow the command anyway), but your suggestion makes sense, thank you.